### DISCUSSION TITLE
How to call function from callback_data that changes?

### DISCUSSION THREAD

**Discussion by ArthurPains** [python-telegram-bot/python-telegram-bot#3683] [Q&A] [answered]
I'm having a problem when I need to deal with InlineKeyboardButton that change its callback_data constantly.

What I want is that after I select one of the buttons with the name of the games (using edit_game function), I call another function (def select_side), but how do I pass this to the CallbackQueryHandler(select_side, pattern="^" + **(I DON'T KNOW WHAT PUT HERE)** + "$ ") ?

I've tried using loop in every possible way to create a pattern for each game but I couldn't. How can I do this?

I took the inlinekeyboard2.py example and tried to implement my code from it:

```
async def edit_game(update: Update, context: ContextTypes.DEFAULT_TYPE) -> None:
    query = update.callback_query
    await query.answer()
    games_name = list(games.keys())
    reply_markup = InlineKeyboardMarkup([[InlineKeyboardButton(game, callback_data=game)] for game in games_name])
    await query.edit_message_text(text='Choose the game you want to edit:', reply_markup=reply_markup)

def main() -> None:
    """Run the bot."""
    application = Application.builder().token("TOKEN").build()

    conv_handler = ConversationHandler(
        entry_points=[CommandHandler("start", start)],
        states={
            START_ROUTES: [
                CallbackQueryHandler(edit_game, pattern="^" + str(EDIT_GAME) + "$"),
            ],               
        },
        fallbacks=[CommandHandler("start", start)],
    )
    application.add_handler(conv_handler)
    application.run_polling()


if __name__ == "__main__":
    main()
```

**Comment by Bibo-Joshi** [comment, marked as answer]
The `pattern` argument is optional. You could just omit it.

**Reply by ArthurPains** [reply]
Dude it worked perfectly! Thank you so much, I didn't think it was something simple lol
